spec/system/authentication/user_logs_in_with_email_spec.rb

```diff
@@ -13,25 +13,33 @@
     let(:user) { build(:user, saw_onboarding: false) }
 
     context "when using valid credentials" do
+      def sign_up_user
+        visit sign_up_path(state: "new-user")
+        click_link(sign_up_link, match: :first)
+
+        fill_in_user(user)
+        click_button("Sign up", match: :first)
+      end
+
       it "creates a new user", js: true do
         expect do
-          visit sign_up_path(state: "new-user")
-          click_link(sign_up_link, match: :first)
-
-          fill_in_user(user)
-          click_button("Sign up", match: :first)
+          sign_up_user
         end.to change(User, :count).by(1)
       end
 
       it "logs in and redirects to email confirmation" do
-        visit sign_up_path(state: "new-user")
-        click_link(sign_up_link, match: :first)
-
-        fill_in_user(user)
-        click_button("Sign up", match: :first)
+        sign_up_user
 
         expect(page).to have_current_path("/confirm-email", ignore_query: true)
       end
+
+      it "displays the properly decoded email" do
+        decoded_email = user.email.sub("@", "+something@")
```

Comment:
`CGI.unescape` ?

Reply:
Hmm sorry? `decoded_email` is to mock an email address that has the `+` in it, and `CGI.unescape` would remove that.

Reply:
ah sorry!